dotCMS/src/integration-test/java/com/dotcms/filters/VanityUrlFilterTest.java

```diff
@@ -0,0 +1,180 @@
+     * In this case, we will request a url from dotcms and check to see that we get the results from dotcms.com
+     */
+
+    @Test
+    public void test_that_vanity_url_filter_handles_proxy_requests() throws Exception {
+
+
+        String title = "VanityURL"  + System.currentTimeMillis();
+        String site = defaultHost.getIdentifier();
+        String uri = "/testing_forward" + System.currentTimeMillis();
+        String forwardTo = "https://dotcms.com";
+        int action = 200;
+        int order = 1;
+
+        Contentlet contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
+        filtersUtil.publishVanityUrl(contentlet1);
+
+        final HttpServletRequest request = new MockHttpRequest(defaultHost.getHostname(), uri).request();
+  
+        final File tmp = File.createTempFile("testingVanity", "test");
+        tmp.deleteOnExit();
+        
+        final HttpServletResponse response = new MockHttpCaptureResponse(new MockHttpResponse().response(), tmp).response();
+
+        final VanityURLFilter filter = new VanityURLFilter();
+        
+        filter.doFilter(request, response, null);
+        
+        assert(tmp.exists());
+        String content = FileUtil.read(tmp);
+        assert(content!=null);
+        assert(content.contains("All rights reserved"));
+        assert(content.contains("<meta property=\"og:url\" content=\"https://dotcms.com/\">"));
+        
+
+    }
+
+    /**
+     * this tests that vanity url redirects work as expected
+     * @throws Exception
+     */
+    @Test
+    public void test_that_vanity_url_filter_handles_redirects() throws Exception {
+
+
+        String title = "VanityURL"  + System.currentTimeMillis();
+        String site = defaultHost.getIdentifier();
+        String uri = "/testing_301" + System.currentTimeMillis();
+        String forwardTo = "https://dotcms.com";
+        int action = 301;
+        int order = 1;
+
+        Contentlet contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
+        filtersUtil.publishVanityUrl(contentlet1);
+
+
+        final HttpServletRequest request = new MockHttpRequest(defaultHost.getHostname(), uri).request();
+        final HttpServletResponse response = new MockHttpStatusResponse(new MockHeaderResponse(new MockHttpResponse().response()).response()).response();
+
+        
+        VanityURLFilter filter = new VanityURLFilter();
+        
+        filter.doFilter(request, response, null);
+
+        Collection<String> list=  response.getHeaderNames();
+        assert(response.getHeader("Location").equals(forwardTo));
+        assert(response.getStatus()==301);
+
+    }
+
+    /**
+     * this tests that vanity urls still work after clearing the vanity url cache
+     * @throws Exception
+     */
+    @Test
+    public void test_cache_flush_does_not_break_vanities() throws Exception {
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)